Source/ACE/Network/GameMessages/Messages/GameMessageMotion.cs

```diff
@@ -13,40 +17,46 @@ public GameMessageMotion()
 
         public GameMessageMotion(Entity.WorldObject animationTarget, Session session, MotionCommand command, float speed = 1.0f)
             : this()
         {
-            WriteBase(animationTarget, session, MotionActivity.Idle, MotionType.General, MotionFlags.None, MotionStance.Standing);
+            WriteBase(animationTarget, session, MotionActivity.Idle, MovementTypes.Invalid, MotionFlags.None, MotionStance.Standing);
             List<MotionItem> animations = new List<MotionItem>() { new MotionItem(command, speed) };
             WriteAnimations(animationTarget, animations);
         }
 
-        private void WriteBase(Entity.WorldObject animationTarget, Session session, MotionActivity activity,
-            MotionType type, MotionFlags flags, MotionStance stance)
+        private void WriteBase(WorldObject animationTarget, Session session, MotionActivity activity,
+            MovementTypes type, MotionFlags flags, MotionStance stance)
         {
-            Writer.WriteGuid(animationTarget.Guid);
-            Writer.Write((ushort)session.Player.TotalLogins);
-            Writer.Write(animationTarget.Sequences.GetNextSequence(Sequence.SequenceType.MotionMessage));
-            Writer.Write((ushort)1); // Index, needs more research, it changes sometimes, but not every packet
-            Writer.Write((ushort)activity);
-            Writer.Write((byte)type);
-            Writer.Write((byte)flags);
-            Writer.Write((ushort)stance);
+            Writer.WriteGuid(animationTarget.Guid); // Object_Id (uint)
+            Writer.Write((ushort)session.Player.TotalLogins); // Instance_Timestamp
+            Writer.Write(animationTarget.Sequences.GetNextSequence(SequenceType.MotionMessage)); // Movement_Timestamp
+            Writer.Write(animationTarget.Sequences.GetNextSequence(SequenceType.ServerControl)); // Server_Control_Timestamp
+            // The order of these is in question between the movement and server control.   Looks like in the client this is the right order
+            // in AC Log View it is reversed.
+            // TODO: reseach the correct order.   We can probably get away with either for now.
+            Writer.Write((ushort)type); // movement_type
+            Writer.Write((byte)activity); // autonomous flag - 1 or 0.   I think this is set if you have are holding the run key or some other autonomous movement            
+            Writer.Write((byte)flags); // these can be or and has sticky object | is long jump mode |
+            Writer.Write((ushort)stance); // called command in the client
+
         }
 
-        private void WriteAnimations(Entity.WorldObject animationTarget, List<MotionItem> items)
+        private void WriteAnimations(WorldObject animationTarget, List<MotionItem> items)
         {
-            uint generalFlags = (uint)items.Count << 7;
-            Writer.Write((uint)generalFlags);
+            if (animationTarget == null) throw new ArgumentNullException(nameof(animationTarget));
```

Comment:
throwing here is probably not the best idea.  should probably just make a log warning entry and return.